src/main.rs

```diff
@@ -512,44 +521,43 @@ fn run() -> Result<(), Error> {
     if let Some(index) = pbf_relations {
         let index: Vec<_> = index.collect();
 
-        info!("Building relations index...");
-
         // We need to build the index of relation ids first, since relations can refer
         // again to relations.
-        let relations_id_to_idx = build_relations_index(&mut file, index.iter())?;
-        info!("Relations index built.");
-
-        let mut pb = ProgressBar::new(index.len() as u64);
-        pb.message("Converting relations...");
+        let relations_id_to_idx = build_relations_index(&args.input, index.clone().into_iter())?;
 
         let mut relations = builder.start_relations()?;
         relations.grow()?; // index 0 is reserved for invalid relation
 
         let mut relation_members = builder.start_relation_members()?;
         relation_members.grow()?; // index 0 is ALSO reserved for invalid relation
 
-        for idx in index {
-            let block: osmpbf::PrimitiveBlock = read_block(&mut file, &idx)?;
-            stats += serialize_relations(
-                &block,
-                &nodes_id_to_idx,
-                &ways_id_to_idx,
-                &relations_id_to_idx,
-                &mut stringtable,
-                &mut relations,
-                &mut relation_members,
-                &mut tags,
-            )?;
-            pb.inc();
-        }
+        parallel::parallel_process(
+            "Converting relations...",
```

Comment:
I think the progress bar and the message can be just initialized before parallel process. The progress bar can be increased in the consume closure.

Reply:
Not as compact, but possible. Normally I'd like to avoid copy/paste, but I can see that keeping logging/output to main.rs can make sense